### ISSUE TITLE
Question: Admin rights in STF needed for adb connect?

### ISSUE THREAD

**Issue by HvandenBerg** [DeviceFarmer/stf#218]
When I log into STF as an admin and start using a device I can connect to this device through the cmd line using adb connect and I can see the conected device with adb devices

When I log in with a non-admin account and I try to connect to the device I get a messagethat I cannot connect, 10061 connection refused and when I use the command adb devices the device does appear in the list but as offline

Do you need admin rights to use this functionality??

**Comment by jachstet-sea** [comment]
Hi Hugo. Every machine running adb has a unique key pair. The public key needs to be configured in your STF-account in order to allow the machine to connect.
Can you please check that you have the correct ADB key associated with the non-admin account?

**Comment by HvandenBerg** [comment]
@jachstet-sea Thank you! that was indeed the issue. I added the keys for myself and collegues with the admin account assuming then that other users could also use them. Never assume :-) I'll tell them to add the keys under their own accounts

**Comment by arist0v** [comment]
Hello, i have similar issue:

i take owner ship of the device with the api (then it become with the flag stop automation)

then if i try to connect, i got 
```
ConnectionRefusedError: [Errno 111] Connection refused

```
my adb keys is added to the server so i can'T figure why it didn'T work any help would be welcome

**Comment by denis99999** [comment]
@arist0v ensure you ran the right API calls sequence before to ADB connect to your device:

1. takes control of the device : calling `API POST /user/devices/{serial}`
2. opens the device worker listen port and gets the Remote debug URL : calling `API POST /user/devices/{serial}/remoteConnect`
3. ADB connects to the device using the previous Remote debug URL

Ensure also you the token and the ADB key you use have been created under the same user account, remember also an ADB key can't be registered under two distinct user accounts.

Then it should works! If it is not the case, probably your connection error is due to a network problem not related to STF (i.e. FW, etc.)

**Comment by arist0v** [comment]
This is it, i didn'T do the remoteConnect since i was getting the url from the device information,

It should have some improvement in the documentation, maybe getting a website up again and request help from community, i could do some stuff if you wish

**Comment by denis99999** [comment]
> his is it, i didn'T do the remoteConnect since i was getting the url from the device information,

The remoteConnect API call must be made, otherwise the listening port of the running device is not open and therefore an ADB connection cannot be opened!



> It should have some improvement in the documentation, maybe getting a website up again and request help from community, i could do some stuff if you wish

Yes you are welcome!

**Comment by arist0v** [comment]
> Yes you are welcome!

Tell me where i could write article ??? since there is no website or wiki?? and i will gladly write my experience and whatnot related to it to help new user

**Comment by denis99999** [comment]
@arist0v, API usage is already documented in [API.md](https://github.com/DeviceFarmer/stf/blob/master/doc/API.md) and [GroupFeature.pdf at Section 7](https://github.com/DeviceFarmer/stf/blob/master/doc/GroupFeature.pdf), so if you want to add some improvements in the documentation, I think you can add or update some markdown files (.md) in `/doc` folder of the Devicefarmer/stf project, what do you think of this @koral-- ?

**Comment by arist0v** [comment]
@denis99999 shouldn'it be better to, at some point, have a website for the documentation to help people find answer on google?

**Comment by denis99999** [comment]
@arist0v , yes it would be better and you are welcome if you want to build it entirely, and especially see with Koral what he thinks about it because for my part I don't really have the time to take care of it, sorry.
